npm/pkg/dataplane/ipsets/ipsetmanager.go

```diff
@@ -170,108 +171,209 @@ func (iMgr *IPSetManager) DeleteReference(setName, referenceName string, referen
 }
 
 func (iMgr *IPSetManager) AddToSets(addToSets []*IPSetMetadata, ip, podKey string) error {
-	// check if the IP is IPV4 family in controller
+	if len(addToSets) == 0 {
+		return nil
+	}
+	// TODO check if the IP is IPV4 family in controller
 	iMgr.Lock()
 	defer iMgr.Unlock()
 
-	if err := iMgr.checkForIPUpdateErrors(addToSets, npmerrors.AppendIPSet); err != nil {
-		return err
-	}
+	for _, metadata := range addToSets {
+		// 1. check for errors and create a missing set
+		prefixedName := metadata.GetPrefixName()
+		set, exists := iMgr.setMap[prefixedName]
+		if !exists {
+			// NOTE: any newly created IPSet will still be in the cache if an error is returned later
+			iMgr.createIPSet(metadata)
+			set = iMgr.setMap[prefixedName]
+		}
+		if set.Kind != HashSet {
+			return npmerrors.Errorf(npmerrors.AppendIPSet, false, fmt.Sprintf("ipset %s is not a hash set", prefixedName))
+		}
 
-	for _, setMetadata := range addToSets {
-		prefixedName := setMetadata.GetPrefixName()
-		set := iMgr.setMap[prefixedName]
-		cachedPodKey, ok := set.IPPodKey[ip]
+		// 2. add ip to the set, and update the pod key
+		_, ok := set.IPPodKey[ip]
 		set.IPPodKey[ip] = podKey
-		if ok && cachedPodKey != podKey {
-			klog.Infof("AddToSet: PodOwner has changed for Ip: %s, setName:%s, Old podKey: %s, new podKey: %s. Replace context with new PodOwner.",
-				ip, set.Name, cachedPodKey, podKey)
+		if ok {
 			continue
 		}
 
-		iMgr.modifyCacheForKernelMemberUpdate(prefixedName)
+		iMgr.modifyCacheForKernelMemberUpdate(set)
 		metrics.AddEntryToIPSet(prefixedName)
```

Comment:
We do not have to check this and print a info message, we can ignore mismatched podKey in add. Wdyt ?